faker/providers/ssn/en_US/__init__.py

```diff
@@ -1,19 +1,172 @@
 # coding=utf-8
 from __future__ import unicode_literals
+
+import random
 from .. import Provider as BaseProvider
 
 
 class Provider(BaseProvider):
+    SSN_TYPE = 'ssn'
+    ITIN_TYPE = 'itin'
+    EIN_TYPE = 'ein'
+
+    def itin(self):
+        """Generate a random United States Individual Taxpayer Identification Number (ITIN).
+
+        An United States Individual Taxpayer Identification Number
+        (ITIN) is a tax processing number issued by the Internal
+        Revenue Service. It is a nine-digit number that always begins
+        with the number 9 and has a range of 70-88 in the fourth and
+        fifth digit. Effective April 12, 2011, the range was extended
+        to include 900-70-0000 through 999-88-9999, 900-90-0000
+        through 999-92-9999 and 900-94-0000 through 999-99-9999.
+        https://www.irs.gov/individuals/international-taxpayers/general-itin-information
+        """
+
+        area = self.random_int(min=900, max=999)
+        group = self.random_int(min=70, max=99)
+        serial = self.random_int(min=0, max=9999)
 
-    def ssn(self):
-        # Certain numbers are invalid for U.S. SSNs. The area (first 3 digits)
-        # cannot be 666 or 900-999. The group number (middle digits) cannot be
-        # 00. The serial (last 4 digits) cannot be 0000
-        area = self.random_int(min=1, max=899)
-        if area == 666:
-            area += 1
-        group = self.random_int(1, 99)
-        serial = self.random_int(1, 9999)
+        # The group number must be between 70 and 99 inclusively but not 89 or 93
+        group = random.choice([x for x in range(70,100) if x not in [89, 93]])
```

Comment:
What's the reason for changing the SSN generation rules? Were the sets of `ssn` and `itin` numbers overlapping?

Reply:
Right, SSNs and ITINs have different generation/validation rules -- the previous approach didn't generate either consistently: It generated a SSN/ITIN like value that would fail validation checks.